src/core/cwa.ts

```diff
@@ -68,24 +79,27 @@ export default class Cwa {
         .then(resource => ({ resource, path }))
     }, { concurrency: this.options.fetchConcurrency || null })
       .each(({ resource }) => {
-        this.$storage.setResource({ id: resource['@id'], name: resource['@type'], isNew: false, resource })
+        resource && this.$storage.setResource({ id: resource['@id'], name: resource['@type'], isNew: false, resource })
         return callback(resource)
       })
   }
 
   public async fetchRoute (path) {
-    const routeData = await this.fetchItem({ path: `/_/routes/${path}`, preload: ['/page/layout/componentCollections/*/componentPositions/*/component', '/page/componentCollections/*/componentPositions/*/component'] })
-    const pageData = await this.fetchItem({ path: routeData.page })
-    const layoutData = await this.fetchItem({ path: pageData.layout })
+    const { page } = await this.fetchItem({ path: `/_/routes/${path}`, preload: ['/page/layout/componentCollections/*/componentPositions/*/component', '/page/componentCollections/*/componentPositions/*/component'] })
```

Comment:
Shouldn't it be `page = await`?

Reply:
I actually reverted this anyway, page may be null. There's a hierarchal structure I forgot where there can be a `pageData` resource between `route` and `page` - been playing this morning.